buffer/src/test/java/io/netty/buffer/AbstractCompositeByteBufTest.java

```diff
@@ -1642,6 +1642,47 @@ public void execute() {
         }
     }
 
+    @Test
+    public void testOverflowWhileUseConstructorWithOffset() {
+        int capacity = 1024 * 1024; // 1MB
+        final ByteBuf buffer = Unpooled.buffer(capacity).writeZero(capacity);
+        final List<ByteBuf> buffers = new ArrayList<ByteBuf>();
+        for (int i = 0; i >= 0; i += capacity) {
```

Comment:
Loop until `int` overflows? The loop looks correct, but weird. Perhaps it'd be more clear with a `long` loop until capacity is greater than `(long) Integer.MAX_VALUE`.

Reply:
yeah loop until it overflow. But yep I think a long might be "more clean". Let's do it :)